client/test/fluid_ac_test.ml

```diff
@@ -518,5 +520,51 @@ let run () =
                 (acFor ~tlid ~pos:13 m |> filterValid |> List.map ~f:AC.asName)
               |> toEqual ["o"; "Ok"; "Nothing"; "Error"]) ;
           ()) ;
+      describe "filter in DB::query" (fun () ->
+          let isFunction = function FACFunction _ -> true | _ -> false in
+          let isVariable = function FACVariable _ -> true | _ -> false in
+          let isKeyword = function FACKeyword _ -> true | _ -> false in
+          let isConstructor = function
+            | FACConstructorName _ ->
+                true
+            | _ ->
+                false
+          in
+          let isLiteral = function FACLiteral _ -> true | _ -> false in
+          let tlid = TLID.fromString "123" in
+          let expr =
+            fn "DB::query_v4" [str "MyDB"; lambda ["value"] (blank ())]
+          in
+          let m =
+            defaultModel
+              ~tlid
+              ~dbs:[aDB ~tlid:(TLID.fromString "db123") ()]
+              ~handlers:[aHandler ~expr ()]
+              ()
+          in
+          let ac = acFor ~pos:50 m in
+          let valid = filterValid ac in
+          test "includes supported functions only" (fun () ->
+              expect
+                (valid |> List.filter ~f:isFunction |> List.map ~f:AC.asName)
+              |> toEqual ["InQuery::whatever"]) ;
+          test "includes variables" (fun () ->
+              expect
+                ( valid
+                |> List.filter ~f:isVariable
+                |> List.map ~f:AC.asName
+                |> List.filter ~f:(fun n -> n = "value") )
+              |> toEqual ["value"]) ;
+          test "includes variables" (fun () ->
+              expect (valid |> List.filter ~f:isLiteral |> List.map ~f:AC.asName)
+              |> toEqual ["true"; "false"; "null"]) ;
+          test "includes keywords" (fun () ->
+              expect (valid |> List.filter ~f:isKeyword |> List.map ~f:AC.asName)
+              |> toEqual ["let"; "|>"]) ;
+          test "includes constructors" (fun () ->
+              expect
+                (valid |> List.filter ~f:isConstructor |> List.map ~f:AC.asName)
+              |> toEqual ["Just"; "Nothing"; "Ok"; "Error"]) ;
```

Comment:
@pbiggar the last question I have is are these constructors supported by queries/do they make sense in a query?

Reply:
They are not currently supported. It seems they don't have support in `Sql_compiler.partially_evaluate` or in `Sql_compiler.lambda_to_sql` so that means they aren't supported at all.

Reply:
filtered out!  I think this is good to go?

Reply:
Wonderful. I hope to review this on Tuesday!